sds/src/match_validation/http_validator.rs

```diff
@@ -1,20 +1,201 @@
+    pub fn set_method(&mut self, method: HttpMethod) -> &mut Self {
+        self.method = method;
+        self
+    }
+    pub fn set_valid_http_status_code(
+        &mut self,
+        valid_http_status_code: Vec<Range<u16>>,
+    ) -> &mut Self {
+        self.valid_http_status_code = valid_http_status_code;
+        self
+    }
+    pub fn set_invalid_http_status_code(
+        &mut self,
+        invalid_http_status_code: Vec<Range<u16>>,
+    ) -> &mut Self {
+        self.invalid_http_status_code = invalid_http_status_code;
+        self
+    }
+    pub fn set_timeout(&mut self, timeout: Duration) -> &mut Self {
+        self.options.timeout = timeout;
+        self
+    }
+    pub fn build(&self) -> HttpValidatorConfig {
+        HttpValidatorConfig {
+            endpoint: self.endpoint.clone(),
+            method: self.method.clone(),
+            request_header: self.request_header.clone(),
+            valid_http_status_code: self.valid_http_status_code.clone(),
+            invalid_http_status_code: self.invalid_http_status_code.clone(),
+            options: self.options.clone(),
+        }
+    }
+}
+
+pub struct HttpValidatorHelper;
+
+impl HttpValidatorHelper {
+    pub fn new_github_config_builder() -> HttpValidatorConfigBuilder {
+        let mut builder =
+            HttpValidatorConfigBuilder::new("https://api.github.com/octocat".to_string());
+        builder.set_request_header(vec![
+            RequestHeader {
+                key: "Authorization".to_string(),
+                value: "Bearer $MATCH".to_string(),
+            },
+            RequestHeader {
+                key: "User-Agent".to_string(),
+                value: "TEST_DD_SDS".to_string(),
+            },
+            RequestHeader {
+                key: "X-GitHub-Api-Version".to_string(),
+                value: "2022-11-28".to_string(),
+            },
+        ]);
+        builder
+    }
+
+    pub fn new_datadog_config_builder() -> HttpValidatorConfigBuilder {
+        let mut builder = HttpValidatorConfigBuilder::new(
+            "https://api.datadoghq.com/api/v1/validate".to_string(),
+        );
+        builder.set_request_header(vec![
+            RequestHeader {
+                key: "DD-API-KEY".to_string(),
+                value: "$MATCH".to_string(),
+            },
+            RequestHeader {
+                key: "User-Agent".to_string(),
+                value: "TEST_DD_SDS".to_string(),
+            },
+            RequestHeader {
+                key: "Accept".to_string(),
+                value: "application/json".to_string(),
+            },
+        ]);
+        builder
     }
 }
 
 #[async_trait]
 impl MatchValidator for HttpValidator {
-    async fn validate(&self, _: &mut Vec<RuleMatch>, _: &Vec<Box<dyn CompiledRuleDyn>>) {}
+    async fn validate(&self, matches: &mut Vec<RuleMatch>, _: &Vec<Box<dyn CompiledRuleDyn>>) {
+        // Let's reqwest the HTTP API endpoint to validate the matches
+        let client = Client::new();
```

Comment:
This should be able to re-use the same global `Client` that is used in the AWS validator.

Reply:
true will do the same

Reply:
FIXED